backend/de.metas.dunning/src/main/java/de/metas/dunning/api/impl/DunningBL.java

```diff
@@ -247,6 +252,75 @@ public void processCandidates(final IDunningContext context, final IDunningCandi
 		}
 
 		dunningProducer.finish();
+
+		sendMassNotifications(dunningProducer.getCreatedDunningDocIds(), Language.getBaseLanguage());
+	}
+
+	private void sendMassNotifications(@NonNull final Collection<DunningDocId> createdDunningDocIds,
+			@NonNull final Language language)
+	{
+		dunningDAO.getByIdsInTrx(createdDunningDocIds)
+				.stream()
+				.filter(doc -> groupByBoilerPlate(doc) != null)
+				.collect(Collectors.groupingBy(this::groupByBoilerPlate))
+				.forEach((boilerPlateWithLineId, dunningDocs)-> sendMassNotifications(boilerPlateWithLineId,dunningDocs, language));
+	}
+
+	@Nullable
+	private BoilerPlateWithLineId groupByBoilerPlate(final I_C_DunningDoc doc)
+	{
+		final DocTypeId docTypeId = DocTypeId.ofRepoIdOrNull(doc.getC_DocType_ID());
+		if (docTypeId != null)
+		{
+			return docTypeDAO.getById(docTypeId).getMassGenerateNotification();
+		}
+		return null;
+	}
+
+	private void sendMassNotifications(@Nullable final BoilerPlateWithLineId boilerPlateWithLineId,
```

Comment:
pls do a null analysis... param boilerPlateWithLineId is nullable but sendMassNotifications requires it not null

Reply:
made NonNull.